addons/payment_stripe_sca/models/payment.py

```diff
@@ -0,0 +1,281 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import logging
+import requests
+import pprint
+from werkzeug import urls
+
+from odoo import api, models, fields, _
+from odoo.exceptions import ValidationError
+from odoo.tools.safe_eval import safe_eval
+from odoo.addons.payment_stripe_sca.controllers.main import StripeControllerSCA as StripeController
+from odoo.addons.payment_stripe.models.payment import INT_CURRENCIES
+from odoo.tools.float_utils import float_round
+
+_logger = logging.getLogger(__name__)
+
+
+class PaymentAcquirerStripeSCA(models.Model):
+    _inherit = "payment.acquirer"
+
+    def stripe_form_generate_values(self, tx_values):
+        self.ensure_one()
+        base_url = self.env["ir.config_parameter"].sudo().get_param("web.base.url")
+        stripe_session_data = {
+            "payment_method_types[]": "card",
+            "line_items[][amount]": int(
+                tx_values["amount"]
+                if tx_values["currency"].name in INT_CURRENCIES
+                else float_round(tx_values["amount"] * 100, 2)
+            ),
+            "line_items[][currency]": tx_values["currency"].name,
+            "line_items[][quantity]": 1,
+            "line_items[][name]": tx_values["reference"],
+            "client_reference_id": tx_values["reference"],
+            "success_url": urls.url_join(base_url, StripeController._success_url) + "?reference=%s&return_url=%s" % (tx_values["reference"], tx_values.get('return_url')),
+            "cancel_url": urls.url_join(base_url, StripeController._cancel_url) + "?reference=%s&return_url=%s" % (tx_values["reference"], tx_values.get('return_url')),
+            "customer_email": tx_values["partner_email"],
+        }
```

Comment:
We are running in double-charge issues lately and Stripe provides [idempotent requests](https://stripe.com/docs/api/idempotent_requests) to prevent that from happening. 
```python
charge = stripe.Charge.create(
  amount=2000,
  currency="eur",
  description="Charge for jenny.rosen@example.com",
  source="tok_mastercard",
  idempotency_key='G7pKOH4lvy824Pcn' # unique idempotency key
)
```
Would that be something that could be included on this PR?  
@wtaferner has more insights about the issue on our backend.

Reply:
I am not 100% sure, but I think in case of a concurrency retry we will run into this scenario and I guess as @brunojppb says the idempotent request would help to avoid such double payments.

Reply:
As this is the session creation and the charge is created natively by Stripe within the external checkout session, it should be now handled by Stripe instead of Odoo creating the charge itself and potentially redoing the request in case of a retry. It is still true for the old implementation, so the key could help there.